presto-main/src/test/java/io/prestosql/sql/query/TestSubqueries.java

```diff
@@ -119,8 +137,51 @@ public void testCorrelatedSubqueriesWithLimit()
                 "VALUES true",
                 false);
         assertions.assertFails(
-                "SELECT (SELECT t.a FROM (VALUES 1, 2, 3) t(a) WHERE t.a=t2.b ORDER BY a FETCH FIRST ROW WITH TIES) FROM (VALUES 1) t2(b)",
+                "SELECT (SELECT t.a FROM (VALUES 1, 2, 3) t(a) WHERE t.a = t2.b ORDER BY a FETCH FIRST ROW WITH TIES) FROM (VALUES 1) t2(b)",
                 UNSUPPORTED_CORRELATED_SUBQUERY_ERROR_MSG);
+        assertions.assertQuery(
+                "SELECT * " +
+                        "FROM (VALUES 1, 2, 3, null) outer_relation(id) " +
+                        "LEFT JOIN LATERAL " +
+                        "(SELECT value FROM " +
+                        "(VALUES " +
+                        "(1, 'a'), " +
+                        "(1, 'a'), " +
+                        "(1, 'a'), " +
+                        "(1, 'a'), " +
+                        "(2, 'b'), " +
+                        "(null, 'c')) inner_relation(id, value) " +
+                        "WHERE outer_relation.id = inner_relation.id " +
+                        "LIMIT 2) " +
+                        "ON TRUE",
+                "VALUES " +
+                        "(1, 'a'), " +
+                        "(1, 'a'), " +
+                        "(2, 'b'), " +
+                        "(3, null), " +
+                        "(null, null)");
+        // TopN in correlated subquery
+        assertions.assertQuery(
+                "SELECT * " +
+                        "FROM (VALUES 1, 2, 3, null) outer_relation(id) " +
+                        "LEFT JOIN LATERAL " +
+                        "(SELECT value FROM " +
+                        "(VALUES " +
+                        "(1, 'd'), " +
+                        "(1, 'c'), " +
+                        "(1, 'b'), " +
+                        "(1, 'a'), " +
+                        "(2, 'w'), " +
+                        "(null, 'x')) inner_relation(id, value) " +
+                        "WHERE outer_relation.id = inner_relation.id " +
+                        "ORDER BY inner_relation.value LIMIT 2) " +
```

Comment:
add test where ordering scheme is also correlated (e.g `ORDER BY outer_relation.id`)

Reply:
ping, you are still missing test for the case above.

Reply:
It's different case than "outer_relation.id = inner_relation.id"

It should be possible to write such test that passes.

Reply:
Ah, right. You meant  order by outer symbol that is in equality predicate? That won't decorrelate either, and for the same reason as the previous test. I could add a test but it won't bring anything new.